[[ Bug 19882 ]] Restrict SSE compiler flag scope in Skia

runrevmark opened the pull request:
This patch ensures that only the files requiring specific compiler
flags are compiled with the specific compiler flags.

This fixes an issue where we were compiling Skia on Linux with
-msse4.2, meaning that the engine would not run on anything which
did not have SSE 4.2.

runrevmark committed:
[[ Bug 19882 ]] Restrict SSE compiler flag scope in Skia

This patch ensures that only the files requiring specific compiler
flags are compiled with the specific compiler flags.

This fixes an issue where we were compiling Skia on Linux with
-msse4.2, meaning that the engine would not run on anything which
did not have SSE 4.2.

livecodeian commented:
@livecode-vulcan review ok 6fbda2a

livecode-vulcan commented:
💙 review by @livecodeian ok 6fbda2a439366e4180a542cf7e58e72283fd5150

runrevmark commented:
@livecodepanos Please can you make sure this is merged and submodules updated... We might want to re-run a CHECK PR on it to make sure it works, but it is important it goes into dp-8 regardless.

livecodeali commented:
@runrevmark The check PR is here and is failing on Windows, that's why this hasn't been merged
https://github.com/livecode/livecode/pull/5614

runrevmark committed:
[[ Bug 19882 ]] Make sure target_arch is compared with x64

This patch ensures that the target-specific opt targets also use
x64 as a comparison option.
